AmplifyPlugins/Core/AWSPluginsCoreTests/Model/Support/QueryPredicateGraphQLTests.swift

```diff
@@ -147,13 +147,39 @@ class QueryPredicateGraphQLTests: XCTestCase {
           ]
         }
         """
-        let result = try predicate.toGraphQLFilterJSON()
+        let result = try predicate.toGraphQLFilterPrettyPrintedJson()
         XCTAssertEqual(result, expected)
     }
+
+    func testJSONSerializationAndDeserialization() throws {
+        let post = Post.keys
+        let predicate = post.id.eq("id") && post.title.beginsWith("Title")
+        let predicateString = try predicate.toGraphQLFilterString()
+        XCTAssertNotNil(predicateString)
+        let predicateJson = try predicateString.toGraphQLFilterJson()
+        guard let filter = predicateJson["and"] as? [[String: Any]] else {
+            XCTFail("should contain 'and' operation")
+            return
+        }
+        let idPredicate = filter[0]
+        guard let idFilter = idPredicate["id"] as? [String: Any] else {
+            XCTFail("should contain 'id' value")
+            return
+        }
+        XCTAssert(idFilter["eq"] as? String == "id")
+
+        let titlePredicate = filter[1]
+        guard let titleFilter = titlePredicate["title"] as? [String: Any] else {
+            XCTFail("should contain 'title' value")
+            return
+        }
+
+        XCTAssert(titleFilter["beginsWith"] as? String == "Title")
+    }
 }
 
 extension QueryPredicate {
-    func toGraphQLFilterJSON() throws -> String {
+    func toGraphQLFilterPrettyPrintedJson() throws -> String {
```

Comment:
this was to differentiate between the code's toGraphQLFilterJSON and test code's toGraphQLFilterJSON which does pretty print for test case assertion, i think what we can do here is in actual code, have optional options and allow test to leverage the parameter by passing in `options = [.prettyPrinted]`

Reply:
not sure I understand `in actual code, have optional options`. If in tests it's always pretty printed, then no options should be needed. Do the extensions clash? Is that why you need another toJSON function?

Reply:
yes the reason for the different naming in test code is the clashing of method names. i  wanted toGraphQLFilterJSON with pretty print options set because our test relies on asserting against a hand crafted pretty printed version of the graphql filter json. "actual code" i meant the production code